sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -141,13 +138,21 @@ protected AnalyzedRelation visitQuery(Query node, StatementAnalysisContext state
                     node.getOffset().get(),
                     relationAnalysisContext.expressionAnalysisContext())));
             }
-            querySpec.orderBy(analyzeOrderBy(
+
+            ExpressionAnalyzer orderByExpressionAnalyzer = new ExpressionAnalyzer(
+                functions,
+                statementContext.sessionContext(),
+                statementContext.convertParamFunction(),
+                new UnionOrderByFieldProvider(relationAnalysisContext.sources()),
```

Comment:
wouldn't it be enought to just create a new map of sources and use that with the existing FullQualifiedNameFieldProvider ?